### ISSUE TITLE
The asynchronous code send by `Runtime.evaluate` can not be executed in worker threads

### ISSUE THREAD

**Issue by theanarkh** [jasnell/piscina#331]
The asynchronous code send by `Runtime.evaluate` can not be executed in worker threads, the code is as follows.

main.js
```js
const path = require('path');
const { Worker } = require('worker_threads');
const { Session } = require('inspector');
const Piscina = require('piscina');

const session = new Session();
session.connect();

async function testPiscina() {
    new Piscina({
        maxThreads: 1,
        filename: path.resolve(__dirname, 'worker.js'),
        // Comment this out and it will work
        // env: {
        //     ...process.env,
        //    PISCINA_DISABLE_ATOMICS: '1'
        // }
    });
    setInterval(() => {}, 10000);
}

function testWorker() {
    new Worker('setInterval(() => {}, 10000)', { eval: true });
}

// the code `console.log(1)` can not be executed in worker thread
testPiscina();
// the code `console.log(1)` can be executed in worker thread
// testWorker();

session.on('NodeWorker.attachedToWorker', (data) => {
    setTimeout(() => {
        session.post('NodeWorker.sendMessageToWorker', {
            sessionId: data.params.sessionId,
            message: JSON.stringify({
                method: "Runtime.evaluate",
                id: +data.params.sessionId,
                params: {
                        includeCommandLineAPI: true, 
                        expression: `setTimeout(() => {
                            console.log(1);
                        }, 1000);`
                },
            })
        }, (err) => err && console.error(err))
    }, 1000)
});
  
session.on('NodeWorker.receivedMessageFromWorker', (data) => {
    console.log('receivedMessageFromWorker', data);
});

session.on('NodeWorker.detachedFromWorker', (data) => {
    console.log('detachedFromWorker', data);
});

session.post("NodeWorker.enable", { waitForDebuggerOnStart: false }, (err) => {
    err && console.log(err);
 });
```
worker.js
```js
module.exports = () => {};
```

**Comment by metcoder95** [comment]
Can you describe what are you trying to achieve and under which Node version?
Testing your code I was fully able to connect the inspector to the worker and send messages back and forth

**Comment by theanarkh** [comment]
>Can you describe what are you trying to achieve and under which Node version? 

v14.19.3 and 19.6.0. 

>Testing your code I was fully able to connect the inspector to the worker and send messages back and forth

Yeah, but the `console.log(1)` can not be executed.

**Comment by theanarkh** [comment]
I see this comment in code.
```
// This function is entered either after receiving the startup message, or
// when we are done with a task. In those situations, the *only* thing we
// expect to happen next is a 'message' on `port`.
// That call would come with the overhead of a C++ → JS boundary crossing,
// including async tracking. So, instead, if there is no task currently
// running, we wait for a signal from the parent thread using Atomics.wait(),
// and read the message from the port instead of generating an event,
// in order to avoid that overhead.
// The one catch is that this stops asynchronous operations that are still
// running from proceeding. Generally, tasks should not spawn asynchronous
// operations without waiting for them to finish, though.
```

**Comment by theanarkh** [comment]
I think because this code `Atomics.wait(sharedBuffer, common_1.kRequestCountField, lastSeenRequestCount);` make the worker suspend.

**Comment by metcoder95** [comment]
ok, got you to know, thought it was only connecting through the inspector.
Piscina makes two steps once kicking off a thread. The first is to run the initialization part, and park the thread with the `Atomics.wait` using a semaphore that will wake up the thread only if there is any incoming task.

And even though, once the task is finished execution, the thread will go back into idle until a new task comes in.

**Comment by metcoder95** [comment]
> I think because this code `Atomics.wait(sharedBuffer, common_1.kRequestCountField, lastSeenRequestCount);` make the worker suspend.

Yes, exactly

**Comment by theanarkh** [comment]
Thanks!
